src/components/BareMetalHosts/BaremetalHostStatus.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getHostStatus } from '../../selectors/host/selectors';
+import { GenericStatus, GenericError, GenericProgress, GenericSuccess, ValidationError } from './StatusComponents';
+
+// Map status strings to components
+const componentMap = {
+  'validation error': ValidationError,
+  'registration error': GenericError,
+  registering: GenericProgress,
+  inspecting: GenericProgress,
+  discovered: GenericSuccess,
+  OK: GenericSuccess,
+  ready: GenericSuccess,
+  provisioned: GenericSuccess,
+  provisioning: GenericProgress,
+  'preparing to provision': GenericProgress,
+  deprovisioning: GenericProgress,
+  'making host available': GenericProgress,
+};
+
+export const BaremetalHostStatus = ({ host }) => {
+  const hostStatus = getHostStatus(host);
```

Comment:
we found out that for better reusability it is better to decouple status functionality from the UI (e.g. vmStatus is used at many places). Please see utils/status. Could you use this approach here?

Basically it is a function which returns
```js
{
   status, // constant
   ...additionalUsefulData // e.g. message
}
```

Reply:
I agree we could make the textMap and componentMap driven by constants. We can also enhance the getHostStatus selector to return an object as you're describing, which would allow us to include error message etc. and the UI components would not have to reach for it separately.

Would it be ok to do this in a subsequent PR? Quite a lot of work depends on this one so we'd like to get this one in.

Reply:
yes. Also this should go to `utils/status`. Selectors should be kept simplest as possible.

I would rather do it in this PR. Especially because other work depends on it. 
It is not a time consuming task and will lead to (hopefully) less bugs.